src/librustdoc/html/render/print_item.rs

```diff
@@ -954,13 +954,13 @@ fn item_trait(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clean:
     if !t.is_object_safe(cx.tcx()) {
         write_section_heading(
             w,
-            "Object Safety",
+            "Trait-Object Safety",
             "object-safety",
```

Comment:
```suggestion
            "dyn-compatibility"
```

might need to update other references of this ID

Reply:
As a nit, one would probably say that "a trait is dyn compatible" (but one might "have a dyn-compatible trait"), in the same way that we're more likely to hyphenate "a forward-compatible change" than "a change that is forward compatible".

Similarly, I'd probably say "dyn compatibility" unhyphenated, in the same way "forward compatibility" most often is.

Reply:
Yeah, that makes sense (attributive vs. predicative usage). I went for unconditional hyphenation here and in my other "object safe → dyn compatible" PRs for legibility (well, that's slightly subjective I guess).

Apart from personal preference, my more argument goes as follows: If you *skim* source code comments and documentation, a "stray" *dyn* has a good chance of tripping you up (that's my assumption) especially since it's jargon, not everyday language.

I'm fine with conditional hyphenation as you've suggested.

Reply:
Tho note that this specific string should stay hyphenated for unrelated reasons – it represents the element ID.